pkg/helm/chartproxy/repos.go

```diff
@@ -101,6 +101,15 @@ type helmRepoGetter struct {
 
 func (b helmRepoGetter) unmarshallConfig(repo unstructured.Unstructured) (*helmRepo, error) {
 	h := &helmRepo{}
+
+	disabled, _, err := unstructured.NestedBool(repo.Object, "spec", "disabled")
+	if err != nil {
```

Comment:
if `disabled` field does not exist, would this trigger error?

I would not skip unmarshaling if repo is disabled - instead I would add add `disabled` field to `helmRepo` struct, unmarshall to it as usual, and then in `List` method decide based on the value of `disabled` field.

Reply:
@pedjak if `disabled` field isn't found, `unstructured.NestedBool` returns false and `err` would be nil.
ref: https://github.com/openshift/console/blob/117280d206855362c3044b7fca51313492eaff8c/vendor/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/helpers.go#L89